src/api/MetricsApi.ts

```diff
@@ -0,0 +1,96 @@
+import { DateTime } from "luxon";
+import { spWebContext } from "../providers/SPWebContext";
+import { getAPIError } from "./InternalErrors";
+import { sleep } from "./ProcessesApiDev";
+import { UserApiConfig } from "./UserApi";
+
+export interface IMetricsApi {
+  /**
+   * Submits a new IMetricEvent
+   *
+   * @param Title The title of the event, e.g. "Buyer Review"
+   * @param ProcessId The IProcess the event is associated with
+   * @param Reason One of the ReworkReasons if given
+   */
+  submitEvent(
+    Title: string,
+    ProcessID: number,
+    Reason?: string
+  ): Promise<IMetricEvent>;
+}
+
+/**
+ * The interface that represents how a Note is formed when it is submitted and returned from submission.
+ */
+interface IMetricEvent {
+  Id?: number;
+  Title: string;
+  ProcessID: number;
+  Reason?: string;
+  AuthorId?: number;
+  Modified?: string;
+}
+
+export default class MetricsApi implements IMetricsApi {
+  private metricsList = spWebContext.lists.getByTitle("Metrics");
+  private userApi = UserApiConfig.getApi();
```

Comment:
Should this just be moved to the DevApi? -- no reason to declare for regular api, as we don't use it

Reply:
Should the private keyword be added to where the userApi is used in the Dev class?